app/models/seats_reservation.rb

```diff
@@ -3,4 +3,22 @@
 class SeatsReservation < ApplicationRecord
   belongs_to :reservation
   belongs_to :cinema_hall
+  validate :validates_seat_reservation
+
+  def seat_is_taken?
+    SeatsReservation
+      .joins(:reservation)
+      .where(
+        row: row,
+        seat_number: seat_number,
+        reservations: { screening: reservation.screening }
+      )
+      .any?
+  end
+
+  private
+
+  def validates_seat_reservation
+    errors.add(:seat_reservation, 'this seat is already taken') if seat_is_taken?
```

Comment:
We discussed that this validation doesn't make sense at all at the model level. Let me explain why.

0. (initial state). The system doesn't have any `SeatReservation` records.
1. We're creating a new object of `SeatReservation`:  `seat_reservation_1 = SeatReservation.new(row: 'A', seat_number: 6, cinema_hall: cinema_hall, reservation: reservation)`.
2. We're checking this validation: `seat_reservation_1.valid? # => true`
3. We're saving the record: `seat_reservation_1.save`
4. We're checking the validation again: `seat_reservation_1.valid? # => false`
I don't think that you would like to achieve a behaviour like that. You basically will end up always with an invalidated record in the database after saving it to the database. I think that you would like to perform this "checking" before saving in your business process instead of treating it as an invariant.

https://lostechies.com/jimmybogard/2016/04/29/validation-inside-or-outside-entities/ may this will be helpful for you.

Reply:
Sure, I remember, but I just had some issue with using method` seat_is_taken?` inside my reservations controller, so I decided to create a draft PR for now and reach out for some help on Monday :)

Reply:
Refactored, thanks for your help 👍